File: CS/DevExpressMvcApplication1/Models/CustomFormHelper.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;
using DevExpress.Web.ASPxScheduler;
using DevExpress.Web.Mvc;
using DevExpress.XtraScheduler;

namespace DevExpressMvcApplication1.Models {
    public class ModelAppointment {
        public ModelAppointment() { }

        [Required]
        public DateTime StartTime { get; set; }

        [Required]
        public DateTime EndTime { get; set; }

        [Required]
        public string Subject { get; set; }

        public int Status { get; set; }
        public string Description { get; set; }
        public int Label { get; set; }
        public string Location { get; set; }
        public bool AllDay { get; set; }
        public int EventType { get; set; }
        public string RecurrenceInfo { get; set; }
        public string ReminderInfo { get; set; }

        [Required]
        [Display(Name = "Resource")]
        public int OwnerId { get; set; }
        public string CustomInfo { get; set; }
        public int ID { get; set; }

        public bool HasReminder { get; set; }

        public Reminder Reminder { get; set; }

        public TimeSpan TimeBeforeStart {
            get { return Reminder == null ? TimeSpan.FromMinutes(10) : Reminder.TimeBeforeStart; }
        }

        public int CompanyID { get; set; }
        public int ContactID { get; set; }

        public ModelAppointment(CustomAppointment source) {
            if(source != null) {
                StartTime = source.StartTime;
                EndTime = source.EndTime;
                Subject = source.Subject;
                Status = source.Status;
                Description = source.Description;
                Label = source.Label;
                Location = source.Location;
                AllDay = source.AllDay;
                EventType = source.EventType;
                RecurrenceInfo = source.RecurrenceInfo;
                ReminderInfo = source.ReminderInfo;
                OwnerId = source.OwnerId;
                CustomInfo = source.CustomInfo;
                ID = source.ID;
                HasReminder = source.ReminderInfo.Trim() != "";
            }
        }
    }

    public class CustomAppointmentTemplateContainer : AppointmentFormTemplateContainer {
        public CustomAppointmentTemplateContainer(MVCxScheduler scheduler) : base(scheduler) { }

        public string CustomInfo {
            get {
                return Appointment.CustomFields["AppointmentCustomField"] != null ? Appointment.CustomFields["AppointmentCustomField"].ToString() : "";
            }
        }

        public int CompanyID {
            get {
                return Appointment.CustomFields["AppointmentCompany"] != null ? Convert.ToInt32(Appointment.CustomFields["AppointmentCompany"]) : 0;
            }
        }

        public int ContactID {
            get {
                return Appointment.CustomFields["AppointmentContact"] != null ? Convert.ToInt32(Appointment.CustomFields["AppointmentContact"]) : 0;
            }
        }

    }
}

File: VB/DevExpressMvcApplication1/Models/CustomFormHelper.vb
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel.DataAnnotations
Imports System.Linq
Imports System.Web

Namespace DevExpressMvcApplication1.Models
    Public Class ModelAppointment
        Public Sub New()
        End Sub

        <Required> _
        Public Property StartTime() As Date

        <Required> _
        Public Property EndTime() As Date

        <Required> _
        Public Property Subject() As String

        Public Property Status() As Integer
        Public Property Description() As String
        Public Property Label() As Integer
        Public Property Location() As String
        Public Property AllDay() As Boolean
        Public Property EventType() As Integer
        Public Property RecurrenceInfo() As String
        Public Property ReminderInfo() As String

        <Required, Display(Name := "Resource")> _
        Public Property OwnerId() As Integer
        Public Property CustomInfo() As String
        Public Property ID() As Integer

        Public Property HasReminder() As Boolean

        Public Property Reminder() As Reminder

        Public ReadOnly Property TimeBeforeStart() As TimeSpan
            Get
                Return If(Reminder Is Nothing, TimeSpan.FromMinutes(10), Reminder.TimeBeforeStart)
            End Get
        End Property

        Public Property CompanyID() As Integer
        Public Property ContactID() As Integer

        Public Sub New(ByVal source As CustomAppointment)
            If source IsNot Nothing Then
                StartTime = source.StartTime
                EndTime = source.EndTime
                Subject = source.Subject
                Status = source.Status
                Description = source.Description
                Label = source.Label
                Location = source.Location
                AllDay = source.AllDay
                EventType = source.EventType
                RecurrenceInfo = source.RecurrenceInfo
                ReminderInfo = source.ReminderInfo
                OwnerId = source.OwnerId
                CustomInfo = source.CustomInfo
                ID = source.ID
                HasReminder = source.ReminderInfo.Trim() <> ""
            End If
        End Sub
    End Class

    Public Class CustomAppointmentTemplateContainer
        Inherits AppointmentFormTemplateContainer

        Public Sub New(ByVal scheduler As MVCxScheduler)
            MyBase.New(scheduler)
        End Sub

        Public ReadOnly Property CustomInfo() As String
            Get
                Return If(Appointment.CustomFields("AppointmentCustomField") IsNot Nothing, Appointment.CustomFields("AppointmentCustomField").ToString(), "")
            End Get
        End Property

        Public ReadOnly Property CompanyID() As Integer
            Get
                Return If(Appointment.CustomFields("AppointmentCompany") IsNot Nothing, Convert.ToInt32(Appointment.CustomFields("AppointmentCompany")), 0)
            End Get
        End Property

        Public ReadOnly Property ContactID() As Integer
            Get
                Return If(Appointment.CustomFields("AppointmentContact") IsNot Nothing, Convert.ToInt32(Appointment.CustomFields("AppointmentContact")), 0)
            End Get
        End Property

    End Class
End Namespace